cmd/dp-search-api/main.go

```diff
@@ -60,10 +63,25 @@ func main() {
 		}
 	}
 
+	elasticHTTPClient := dphttp.NewClient()
 	elasticSearchClient := elasticsearch.New(cfg.ElasticSearchAPIURL, dphttp.NewClient(), cfg.SignElasticsearchRequests, esSigner, cfg.AwsRegion, cfg.AwsService)
 	transformer := transformer.New()
+	svcList := service.NewServiceList(&service.Init{})
 
-	if err := api.CreateAndInitialise(cfg.BindAddr, queryBuilder, elasticSearchClient, transformer, apiErrors); err != nil {
+	// Get HealthCheck
+	ctx := context.Background()
+	hc, err := svcList.GetHealthCheck(cfg, BuildTime, GitCommit, Version)
+
+	if err != nil {
+		log.Event(nil, "could not instantiate healthcheck", log.FATAL, log.Error(err))
+		os.Exit(1)
+	}
+	if err := registerCheckers(ctx, cfg, elasticHTTPClient, esSigner); err != nil {
```

Comment:
Question:  Just making sure I am not missing something here but the `registerCheckers` function returns a `*healthcheck.HealthCheck` not an error so just checking this logic is correct?

If that is the case then you can remove this error check block as `registerCheckers` will log and exit if any errors occur.

Reply:
Yes, you are correct.